### PR TITLE
Add size settings to TextBlock

### PR DESCRIPTION
**Pull request by gustaveen**
## Describe your changes
- Refactor Text component to be able to render rich text content
- Update gray color palette
- Add font size settings. I used datasources for this, that way we can define it once and use in multiple places, neat IMO :) 

## Justify why they are needed
Different font sizes are needed at different breakpoints. Since there will be different needs depending on context we can't just use a single size (like `sm`) and use a predefined size scale.

Screenshot 2023-01-16 at 17 21 46

Screenshot 2023-01-16 at 17 21 28

Screenshot 2023-01-16 at 17 49 55

## Checklist before requesting a review

- [x] I have performed a self-review of my code

### CODE REVIEW

**Comment by gustaveen** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1352** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1352" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1352?utm_source=stack-comment).

File: apps/store/src/blocks/TextBlock.tsx [hunk 1]
```diff
@@ -1,26 +1,29 @@
-import styled from '@emotion/styled'
 import { ISbRichtext, renderRichText, storyblokEditable } from '@storyblok/react'
 import { useMemo } from 'react'
-import { Space, UIColors, getColor } from 'ui'
+import { FontSizes } from 'ui/src/lib/theme/typography'
```
Comment (alebedev) [inline apps/store/src/blocks/TextBlock.tsx:3]: Can it be imported from 'ui' ?

File: apps/store/src/blocks/TextBlock.tsx [hunk 2]
```diff
@@ -1,26 +1,29 @@
...
 
 export const TextBlock = ({ blok }: TextBlockProps) => {
   const contentHtml = useMemo(() => renderRichText(blok.body), [blok.body])
   return (
     <Text
       {...storyblokEditable(blok)}
+      as="div"
       color={blok.color ?? 'textPrimary'}
       y={1}
+      size={{ _: blok.fontSize ?? 'md', ...(blok.fontSizeDesktop && { lg: blok.fontSizeDesktop }) }}
```
Comment (alebedev) [inline apps/store/src/blocks/TextBlock.tsx:24]: New object every render, can we memo it?

**Review by alebedev — APPROVED** [review]
Great job, especially editing experience!

File: apps/store/src/blocks/TextBlock.tsx (hunk 1 above)
Comment (gustaveen) [reply apps/store/src/blocks/TextBlock.tsx:3]: ah, good catch! 👌

File: apps/store/src/blocks/TextBlock.tsx (hunk 2 above)
Comment (gustaveen) [reply apps/store/src/blocks/TextBlock.tsx:24]: In these cases I always wonder if there's any performance benefit memoizing? :D

**Review by guilhermespopolin — APPROVED** [review]
Well done :)

**Pull request merged**